### PR TITLE
Add Qt6 android support and QtQuick client example

### PR DESCRIPTION
**Pull request by freedbrt**

### CODE REVIEW

File: TestAppExample/.gitignore [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# This file is used to ignore files which are generated
```
Comment (frankosterfeld) [inline TestAppExample/.gitignore:1]: I think this could be merged into the top-level .gitignore?

File: TestAppExample/keychainclass.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    m_deleteCredentialJob.setAutoDelete(false);
+}
+
+void KeyChainClass::readKey(const QString &key)
+{
+    m_readCredentialJob.setKey(key);
+
+    QObject::connect(&m_readCredentialJob, &QKeychain::ReadPasswordJob::finished, [=](){
+        if (m_readCredentialJob.error()) {
+            emit error(QString("Write key failed: %1").arg(qPrintable(m_readCredentialJob.errorString())));
```
Comment (frankosterfeld) [inline TestAppExample/keychainclass.cpp:22]: "Write key failed" should read "Read key failed". Also use QStringLiteral() or maybe better, tr() (although nobody will ever translate this test app, I guess)

**Review by frankosterfeld — CHANGES_REQUESTED** [review]

**Review by frankosterfeld — COMMENTED** [review]
Android/Qt6 support patch looks good!

File: TestAppExample/TestAppExample.pro [hunk 3]
```diff
@@ -0,0 +1,29 @@
+QT += quick
+
+CONFIG += c++11
+
+# You can make your code fail to compile if it uses deprecated APIs.
+# In order to do so, uncomment the following line.
+#DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0x060000    # disables all the APIs deprecated before Qt 6.0.0
+
+include(../qt5keychain.pri)
```
Comment (frankosterfeld) [inline TestAppExample/TestAppExample.pro:9]: in current master, this file is now renamed to qtkeychain.pri

**Comment by freedbrt** [comment]
Ok, done

**Comment by PeterPetrik** [comment]
We are migrating merginmaps.com to Qt6, this patch is critical so we can build Android. Would it be possible to merge it and create new qtkeychain release?

**Comment by PeterPetrik** [comment]
This needs a change, https://github.com/frankosterfeld/qtkeychain/blob/f197cdb935b0cfd9881fdc6860874cb8379d1238/CMakeLists.txt#L98 should be removed, there is no Qt6AndroidExtras package at all.

**Comment by PeterPetrik** [comment]
@frankosterfeld shall we fix this PR, or shall I create a simpler PR with just a required changes?

**Comment by frankosterfeld** [comment]
I'd say fix up this PR. I hope i get to a new release when back from vacation.

**Comment by PeterPetrik** [comment]
@frankosterfeld cannot modify his branch, maybe the stuff can be removed after mergin it?

here is the patch
```
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,11 +91,11 @@ else()
   find_package(Qt6 COMPONENTS Core REQUIRED)
   set(QTKEYCHAIN_VERSION_INFIX 6)
 
-  if(ANDROID)
-    find_package(Qt6 COMPONENTS AndroidExtras REQUIRED)
-    include_directories(${Qt6AndroidExtras_INCLUDE_DIRS})
-    set(QTANDROIDEXTRAS_LIBRARIES ${Qt6AndroidExtras_LIBRARIES})
-  endif()
+  # if(ANDROID)
+  #  find_package(Qt6 COMPONENTS AndroidExtras REQUIRED)
+  #  include_directories(${Qt6AndroidExtras_INCLUDE_DIRS})
+  #  set(QTANDROIDEXTRAS_LIBRARIES ${Qt6AndroidExtras_LIBRARIES})
+  # endif()
 
   if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT HAIKU)
     find_package(Qt6 COMPONENTS DBus REQUIRED)
```

**Comment by freedbrt** [comment]
@PeterPetrik i can modify, if it needed

**Comment by PeterPetrik** [comment]
@freedbrt thanks!

**Comment by frankosterfeld** [comment]
Thanks!

**Comment by michaelk83** [comment]
> I hope i get to a new release when back from vacation.

@frankosterfeld , if you're preparing a new release, would it be possible to also include #75 ? It needs an update, but it's a small change, so shouldn't be too complicated...

**Pull request merged**